From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew den Exter <andrew.den.exter@qinetic.com.au>
Date: Tue, 17 Aug 2021 00:48:01 +0000
Subject: [PATCH] Always insert timestamps into the database as strings.

---
 .../core/tracker-data-update.c                 | 18 ++++--------------
 1 file changed, 4 insertions(+), 14 deletions(-)

diff --git a/src/libtracker-sparql/core/tracker-data-update.c b/src/libtracker-sparql/core/tracker-data-update.c
index 5f97d5fae5db63cfe05bc1d794c5446122d99931..f10d96e8104ca6bd8ae8b5b73a1c9ecf8c276875 100644
--- a/src/libtracker-sparql/core/tracker-data-update.c
+++ b/src/libtracker-sparql/core/tracker-data-update.c
@@ -1073,21 +1073,11 @@ statement_bind_gvalue (TrackerDBStatement *stmt,
 	default:
 		if (type == G_TYPE_DATE_TIME) {
 			GDateTime *datetime = g_value_get_boxed (value);
+			gchar *str;
 
-			/* If we have anything that prevents a unix timestamp to be
-			 * lossless, we use the ISO8601 string.
-			 */
-			if (g_date_time_get_utc_offset (datetime) != 0 ||
-			    g_date_time_get_microsecond (datetime) != 0) {
-				gchar *str;
-
-				str = tracker_date_format_iso8601 (datetime);
-				tracker_db_statement_bind_text (stmt, idx, str);
-				g_free (str);
-			} else {
-				tracker_db_statement_bind_int (stmt, idx,
-				                               g_date_time_to_unix (datetime));
-			}
+			str = tracker_date_format_iso8601 (datetime);
+			tracker_db_statement_bind_text (stmt, idx, str);
+			g_free (str);
 		} else if (type == G_TYPE_BYTES) {
 			GBytes *bytes;
 			gconstpointer data;
